### DISCUSSION TITLE
Type error when assinging field error message to react node

### DISCUSSION THREAD

**Discussion by devklick** [react-hook-form/react-hook-form#8874] [Q&A]
I'm trying to implement the [documented example](https://github.com/react-hook-form/resolvers#zod) of react-hook-resolvers with zod, using typescript, however am getting a type error:

```ts
{errors.name?.message && <p>{errors.name?.message}</p>}
```
>Type 'string | FieldError | Merge<FieldError, FieldErrorsImpl<any>>' is not assignable to type 'ReactNode'

What would be the suggested approach for trying to handle the different types that `message` can apparently be?

**Comment by bluebill1049** [comment]
we don't support JSX with messages.

**Reply by devklick** [reply]
Have I completely misunderstood the example in the readme then (the one that I linked in the description)? It looks like it's suggesting that the `message` is OK to be used as a JSX element :thinking:

**Reply by Moshyfawn** [reply]
Can you provide a repro sandbox? @bluebill1049 is saying you can't put JSX as the `message`, but I see that you're unable to render message, which is different

**Reply by devklick** [reply]
@Moshyfawn  I've created an [MRP repo here](https://github.com/devklick/zod-hookform-resolver-issue). Strangely, I couldnt replicate the issue in CodeSandbox, despite using all the same package versions & same tsconfig. I've included the [sanbox here](https://codesandbox.io/s/react-typescript-forked-rlwpv7?file=/src/App.tsx) for reference.

In the meantime, I've got around this error by simply calling `.toString()`. It does the job for now, but I'll probably see some unexpected values when `message` is not actually a string.

**Reply by Moshyfawn** [reply]
Hm.. try putting your form type as a `useForm` generic
```ts
const schema = z.object({ ... })

type FormValues = z.infer<typeof schema>

useForm<FormValues>()
```

**Reply by devklick** [reply]
That works! I honestly could have sworn I tried that already, but I guess not. Thanks for your help.
